protocol/x/clob/e2e/short_term_orders_test.go

```diff
@@ -607,3 +607,370 @@ func TestPlaceOrder(t *testing.T) {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order,
+						},
+					},
+				},
+			},
+		},
+		"Success: Replacement order after partial match in same block": {
+			blocks: []blockOrdersAndExpectations{
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20,
+						*clobtypes.NewMsgPlaceOrder(MustScaleOrder(
+							clobtypes.Order{
+								OrderId:      clobtypes.OrderId{SubaccountId: constants.Bob_Num0, ClientId: 0, ClobPairId: 0},
+								Side:         clobtypes.Order_SIDE_SELL,
+								Quantums:     3,
+								Subticks:     10,
+								GoodTilOneof: &clobtypes.Order_GoodTilBlock{GoodTilBlock: 20},
+							},
+							testapp.DefaultGenesis(),
+						)),
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy5_Price10_GTB21.Order,
+							expectedFillAmount:   PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order.Quantums / 2,
+						},
+					},
+				},
+			},
+		},
+		"Success: Replacement order increases size in next block after partial fill": {
+			blocks: []blockOrdersAndExpectations{
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20,
+						*clobtypes.NewMsgPlaceOrder(MustScaleOrder(
+							clobtypes.Order{
+								OrderId:      clobtypes.OrderId{SubaccountId: constants.Bob_Num0, ClientId: 0, ClobPairId: 0},
+								Side:         clobtypes.Order_SIDE_SELL,
+								Quantums:     3,
+								Subticks:     10,
+								GoodTilOneof: &clobtypes.Order_GoodTilBlock{GoodTilBlock: 20},
+							},
+							testapp.DefaultGenesis(),
+						)),
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order,
+							expectedFillAmount:   PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order.Quantums / 2,
+						},
+					},
+				},
+				{
+					ordersToPlace: []clobtypes.MsgPlaceOrder{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy7_Price10_GTB21,
+					},
+					orderIdsExpectations: map[clobtypes.OrderId]orderIdExpectations{
+						PlaceOrder_Alice_Num0_Id0_Clob0_Buy7_Price10_GTB21.Order.OrderId: {
+							shouldExistOnMemclob: true,
+							expectedOrder:        PlaceOrder_Alice_Num0_Id0_Clob0_Buy7_Price10_GTB21.Order,
+							expectedFillAmount:   PlaceOrder_Alice_Num0_Id0_Clob0_Buy6_Price10_GTB20.Order.Quantums / 2,
+						},
+					},
+				},
+			},
+		},
+		"Success: Replacement order decreases size in next block after partial fill": {
```

Comment:
Should we test the case where replacement order decreases size to below the matched partial fill amount?